src/main/java/marquez/common/models/DatasourceName.java

```diff
@@ -0,0 +1,28 @@
+/*
```

Comment:
Is this necessary as part of this PR? Also, we're starting to head down a path of every field in the database which requires some validation gets a concrete Java class. While I like the typing, I also think this can get a bit out of hand.

Reply:
Added `DatasourceName` to allow for **DatasourceUrn.from**(_DatasourceType_, _DatasourceName_), enabling us to use the type when creating a new datasource URN.

Good point. We really aren't doing much validation for the types we have defined, specifically if they are just wrapped strings. So, agree not much value is added (_currently_) in terms of validation. But, types allow for adding both semantics and context to the params passed to our methods. In the future, types will look more like:

```java
@EqualsAndHashCode
@ToString
public final class DatasourceName {
  @Getter @NonNull private final String value;

  private DatasourceName(@NonNull  final String value) {
    checkNotBlank(value)
    checkArgument(value.length() > 128)
    this.value = value
  }
}
```

Personally, if we are accepting input from callers, we should validate and ensure there's a type defined. But, maybe that's something we should discuss more broadly as a team :)

Reply:
Will be addressed in #404